### PR TITLE
Add 4.17 ODF and OCP config files

### PR DESCRIPTION
**Pull request by petr-balogh**

### CODE REVIEW

File: ocs_ci/framework/conf/ocp_version/ocp-4.17-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+# Config file for nightly OCP 4.16
```
Comment (vavuthu) [inline ocs_ci/framework/conf/ocp_version/ocp-4.17-config.yaml:2]:
```suggestion
# Config file for nightly OCP 4.17
```

File: ocs_ci/framework/conf/ocp_version/ocp-4.17-upgrade.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+# Use this config for upgrading of OCP 4.15 to 4.16 cluster
```
Comment (vavuthu) [inline ocs_ci/framework/conf/ocp_version/ocp-4.17-upgrade.yaml:2]:
```suggestion
# Use this config for upgrading of OCP 4.16 to 4.17 cluster
```

File: ocs_ci/framework/main.py [hunk 3]
```diff
@@ -129,13 +129,12 @@ def process_ocsci_conf(arguments):
         action="store",
         choices=[
             "4.99",
-            "4.10",
-            "4.11",
```
Comment (vavuthu) [inline ocs_ci/framework/main.py:133]: need 4.11 as  for testing upgrade scenarios for 4.12. we need to keep yaml files for 4.11 as well

Comment (petr-balogh) [reply ocs_ci/framework/main.py:133]: we don't run upgrade from 4.10 or 4.11 from master or stable so master/stable branch doesn't need to support n-5 or n-6, we support only n-1 in ocs-ci for now. So for now at least removing some non supported versions from master

**Review by vavuthu — DISMISSED** [review]

**Review by shylesh — APPROVED** [review]
LGTM

**Review by dahorak — APPROVED** [review]

File: ocs_ci/framework/conf/ocp_version/ocp-4.17-config.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  installer_version: "4.17.0-0.nightly"
+  terraform_version: "1.0.11"
+  # ignition_version can be found here
+  # https://docs.openshift.com/container-platform/4.7/post_installation_configuration/machine-configuration-tasks.html#machine-config-overview-post-install-machine-configuration-tasks
+  ignition_version: "3.2.0"
+ENV_DATA:
+  # TODO: replace with 4.17 once template is available
+  vm_template: "rhcos-416.94.202403071059-0-vmware.x86_64"
+  acm_hub_channel: release-2.10
+  acm_version: "2.10"
```
Comment (ebenahar) [inline ocs_ci/framework/conf/ocp_version/ocp-4.17-config.yaml:15]: should be 2.11 for now and 2.11 once we have stable builds

Comment (petr-balogh) [reply ocs_ci/framework/conf/ocp_version/ocp-4.17-config.yaml]: done

**Review by vavuthu — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Pull request merged**